analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -132,53 +158,61 @@ private void analyze(ActivityPayload payload, Set<ActivityCategoryDTO> matchingA
 		// because the lock is per user, it doesn't matter much that we block early.
 		try (LockPool<UUID>.Lock lock = userAnonymizedSynchronizer.lock(payload.userAnonymized.getID()))
 		{
-			analyzeInsideLock(payload, matchingActivityCategories);
+			UserAnonymizedEntityHolder userAnonymizedHolder = new UserAnonymizedEntityHolder(payload.userAnonymized.getID());
+			analyzeInsideLock(userAnonymizedHolder, payload, matchingActivityCategories);
+			if (userAnonymizedHolder.isEntityFetched())
```

Comment:
Can't we rename this to `isEntityUpdated` or `isEntityDirty`? I think not all cases of when the entity is fetched are cases in which the entity is dirty, is it? The method `updateUserAnonymized` is sometimes called already.